### DISCUSSION TITLE
Apple Silicon / M1 optimisation

### DISCUSSION THREAD

**Discussion by chemistr** [mu-editor/mu#1504] [Ideas & New Features]
Hey there

any plans for M1 optimisation? I'm a python beginner and would love if it gets updated soon! And maybe update the icon too so it would fit with the other apps in our dock!

Cheers

**Comment by carlosperate** [comment]
Hi @chemistr,

The Mu Editor project source code in itself is actually a "pure Python" package, meaning is all Python code without any CPU architecture dependent part.

However Mu builds on the shoulders of giants and uses dependencies that do have parts that might not be compatible yet with the M1. In most cases is usually enough to build the packages on the target computer, but the way we currently package everything into an installer means we would ideally need all dependencies to offer Python wheels for the M1 in PyPI as well.

PyQt5 (and related packages), Qscintilla, pygame, and tornado are probably the main packages we would need to track for compatibility with the M1, once all of them offer it we can probably start doing the same.

In the meantime I believe Mu works well with Rosetta, but if you experience any issues it'd be great to hear about them so we can get them resolved :)

> And maybe update the icon too so it would fit with the other apps in our dock!

Not sure I understood this part, does the current icon look particularly out of place in the dock?

**Comment by UniK** [comment]
Is there any update on this issue? 
It has been over two years since the initial question was asked. It is possible that the person who asked the question has grown up without having the opportunity to use this tool. Do the things look better now?
I hope you won't suggest using an emulation layer to run Python again. I assume you don't use it either, do you?
